src/Pulsar.Client/Internal/ServiceNameResolver.fs

```diff
@@ -1,9 +1,24 @@
 ﻿namespace Pulsar.Client.Internal
 
-open Pulsar.Client.Api
+open System
 open System.Net
+open Pulsar.Client.Api
+
+//TODO: implement and move into separate file
+type internal ServiceUri = Uri
+
+type internal ServiceNameResolver(config: PulsarClientConfiguration) =
+
+    member __.GetServiceUrl() = config.ServiceUrl
+
+    member __.UpdateServiceUrl (serviceUrl: string) =
+        let config = { config with ServiceUrl = serviceUrl }
```

Comment:
Most likely original config should be left untouched, but another mutable service address field introduced